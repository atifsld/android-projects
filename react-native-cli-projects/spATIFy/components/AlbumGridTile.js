import React from "react";
import { View, Image, Text, Pressable } from "react-native";
import { StyleSheet } from "react-native";

function AlbumGridTile({album, onPress}) {
    return (
        <Pressable 
            android_ripple={{color: '#ccc'}}
            onPress={onPress}>
            <View style={styles.albumCardView}>
                <View style={styles.albumImageView}>
                    <Image style = {styles.albumArt}
                        source={{uri: album.albumArtUrl}}/>
                </View>
                <View style={styles.albumTextView}>
                    <Text numberOfLines={1} style={styles.albumNameText}>{album.albumName}</Text>
                    <Text numberOfLines={1} style={styles.albumArtistsText}>{album.albumArtists}</Text>
                    <Text numberOfLines={1} style={styles.albumYearText}>{album.albumYear}</Text>
                </View>
            </View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    albumCardView: {
        justifyContent: 'center',
        flex: 1,
        margin: 8,
        height: 260,
        width: 170,
        backgroundColor: "#03DAC6",
        elevation: 8,
        borderRadius: 8,
        shadowColor: 'black',
        shadowOpacity: 0.25,
        shadowOffset: {width: 0, height: 2},
        shadowRadius: 8
    },
    albumArt: {
        height: 160,
        width: 160,
        padding: 5,
        borderRadius: 6
    },
    albumImageView: {
        padding: 5,
        elevation: 12
    },
    albumTextView: {
        padding: 8
    },
    albumNameText: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold',
        numberOfLines: 1,
        ellipsizeMode: 'tail'
    },
    albumArtistsText: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 16,
        color: 'black',
        fontWeight: 'bold'
    },
    albumYearText: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 16,
        color: 'black',
        numberOfLines: 1,
        ellipsizeMode: 'tail'
    }
})

export default AlbumGridTile;