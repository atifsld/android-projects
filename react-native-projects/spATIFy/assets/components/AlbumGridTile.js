import react from "react";
import { View, Image, Text, Pressable } from "react-native";
import { StyleSheet } from "react-native";

function AlbumGridTile(album) {
    return (
        <Pressable android_ripple={{color: '#ccc'}}>
            <View style={styles.albumCardView}>
                <View style={styles.albumImageView}>
                    <Image style = {styles.albumArt}
                        source={{uri: album.album.albumArtUrl}}/>
                </View>
                <View style={styles.albumTextView}>
                    <Text numberOfLines={1} style={styles.albumNameView}>{album.album.albumName}</Text>
                    <Text numberOfLines={1} style={styles.albumArtistsView}>{album.album.albumArtists}</Text>
                    <Text numberOfLines={1} style={styles.albumYearView}>{album.album.albumYear}</Text>
                </View>
            </View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    albumCardView: {
        flex: 1,
        margin: 16,
        height: 250,
        width: 160,
        backgroundColor: "#64FFDA",
        elevation: 8,
        borderRadius: 8,
        shadowColor: 'black',
        shadowOpacity: 0.25,
        shadowOffset: {width: 0, height: 2},
        shadowRadius: 8
    },
    albumArt: {
        height: 150,
        width: 150,
        padding: 5,
        borderRadius: 6
    },
    albumImageView: {
        padding: 5,
        elevation: 12
    },
    albumTextView: {
        padding: 8
    },
    albumNameView: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 20,
        fontWeight: 'bold',
        numberOfLines: 1,
        ellipsizeMode: 'tail'
    },
    albumArtistsView: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 16,
        fontWeight: 'bold'
    },
    albumYearView: {
        paddingHorizontal: 5,
        width: 150,
        fontSize: 16,
        numberOfLines: 1,
        ellipsizeMode: 'tail'
    }
})

export default AlbumGridTile;