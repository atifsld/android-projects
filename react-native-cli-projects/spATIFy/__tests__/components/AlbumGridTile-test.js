import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import AlbumGridTile from '../../components/AlbumGridTile'

const sampleAlbum = {
    "id": "06e00675-980b-4435-be2e-5ec17d942865",
    "albumName": "Head Hunters",
    "albumDescription": "Head Hunters by Herbie Hancock is among the best and most influential jazz-funk albums in jazz history. Recorded and released in the last quarter of 1973, it featured electric synthesizers, elements of funk music and Hancockâ€™s trademark Fender Rhodes,",
    "albumArtUrl": "https://upload.wikimedia.org/wikipedia/en/5/54/Herbie-Hancock-Head-Hunters.png",
    "albumSpotifyUrl": "https://open.spotify.com/album/5fmIolILp5NAtNYiRPjhzA?si=LYzXH1bVQPOS-4lvYcIhgw",
    "albumWikipediaUrl": "https://en.wikipedia.org/wiki/Head_Hunters",
    "albumGeniusUrl": null,
    "albumArtists": "Herbie Hancock",
    "albumLabel": "Columbia",
    "albumTags": "Jazz, Jazz-Funk",
    "albumYear": 1973
}

test('test album grid tile', () => {
    const tree = renderer.create(<AlbumGridTile album={sampleAlbum} />).toJSON();
    expect(tree).toMatchSnapshot();
})