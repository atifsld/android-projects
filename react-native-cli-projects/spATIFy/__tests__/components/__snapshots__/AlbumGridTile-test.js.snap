// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test album grid tile 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#03DAC6",
        "borderRadius": 8,
        "elevation": 8,
        "flex": 1,
        "height": 260,
        "justifyContent": "center",
        "margin": 8,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 8,
        "width": 170,
      }
    }
  >
    <View
      style={
        Object {
          "elevation": 12,
          "padding": 5,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://upload.wikimedia.org/wikipedia/en/5/54/Herbie-Hancock-Head-Hunters.png",
          }
        }
        style={
          Object {
            "borderRadius": 6,
            "height": 160,
            "padding": 5,
            "width": 160,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 8,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "ellipsizeMode": "tail",
            "fontSize": 20,
            "fontWeight": "bold",
            "numberOfLines": 1,
            "paddingHorizontal": 5,
            "width": 150,
          }
        }
      >
        Head Hunters
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingHorizontal": 5,
            "width": 150,
          }
        }
      >
        Herbie Hancock
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "ellipsizeMode": "tail",
            "fontSize": 16,
            "numberOfLines": 1,
            "paddingHorizontal": 5,
            "width": 150,
          }
        }
      >
        1973
      </Text>
    </View>
  </View>
</View>
`;
